### PR TITLE
feat: Export ObjectSchema

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

-   [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request?

Export `ObjectSchema` from the package. I'm working on some changes in the core where I need an instance of `ObjectSchema`, and even though the package is a dependency of `@eslint/config-array`, `@eslint/object-schema` isn't listed in `package.json`, which fails our lint check.

#### What changes did you make? (Give an overview)

Exported `ObjectSchema` as a passthrough from `@eslint/object-schema`.

### CODE REVIEW

**Review by aladdin-add — APPROVED** [review]

**Comment by nzakas** [comment]
@aladdin-add just a reminder, when you approve a PR and don't merge it, please leave a message as to why you didn't merge it.

**Comment by aladdin-add** [comment]
I didn't merge it as I would like another review before merging. :)

**Review by fasttime — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
